pkg/identity/github/principal_test.go

```diff
@@ -0,0 +1,280 @@
+			Claims: map[string]interface{}{
+				"aud":              "sigstore",
+				"exp":              0,
+				"iss":              "https://token.actions.githubusercontent.com",
+				"job_workflow_ref": "sigstore/fulcio/.github/workflows/foo.yaml@refs/heads/main",
+				"ref":              "refs/heads/main",
+				"repository":       "sigstore/fulcio",
+				"sha":              "bf96275471e83ff04ce5c8eb515c04a75d43f854",
+				"sub":              "repo:sigstore/fulcio:ref:refs/heads/main",
+				"workflow":         "foo",
+			},
+			WantErr: true,
+		},
+		`Token missing repository claim should be rejected`: {
+			Claims: map[string]interface{}{
+				"aud":              "sigstore",
+				"event_name":       "push",
+				"exp":              0,
+				"iss":              "https://token.actions.githubusercontent.com",
+				"job_workflow_ref": "sigstore/fulcio/.github/workflows/foo.yaml@refs/heads/main",
+				"ref":              "refs/heads/main",
+				"sha":              "bf96275471e83ff04ce5c8eb515c04a75d43f854",
+				"sub":              "repo:sigstore/fulcio:ref:refs/heads/main",
+				"workflow":         "foo",
+			},
+			WantErr: true,
+		},
+		`Token missing workflow claim should be rejected`: {
+			Claims: map[string]interface{}{
+				"aud":              "sigstore",
+				"event_name":       "push",
+				"exp":              0,
+				"iss":              "https://token.actions.githubusercontent.com",
+				"job_workflow_ref": "sigstore/fulcio/.github/workflows/foo.yaml@refs/heads/main",
+				"ref":              "refs/heads/main",
+				"repository":       "sigstore/fulcio",
+				"sha":              "bf96275471e83ff04ce5c8eb515c04a75d43f854",
+				"sub":              "repo:sigstore/fulcio:ref:refs/heads/main",
+			},
+			WantErr: true,
+		},
+		`Token missing ref claim should be rejected`: {
+			Claims: map[string]interface{}{
+				"aud":              "sigstore",
+				"event_name":       "push",
+				"exp":              0,
+				"iss":              "https://token.actions.githubusercontent.com",
+				"job_workflow_ref": "sigstore/fulcio/.github/workflows/foo.yaml@refs/heads/main",
+				"repository":       "sigstore/fulcio",
+				"sha":              "bf96275471e83ff04ce5c8eb515c04a75d43f854",
+				"sub":              "repo:sigstore/fulcio:ref:refs/heads/main",
+				"workflow":         "foo",
+			},
+			WantErr: true,
+		},
+	}
+
+	for name, test := range tests {
+		t.Run(name, func(t *testing.T) {
+			token := &oidc.IDToken{
+				Issuer:  test.Claims["iss"].(string),
+				Subject: test.Claims["sub"].(string),
+			}
+			claims, err := json.Marshal(test.Claims)
+			if err != nil {
+				t.Fatal(err)
+			}
+			withClaims(token, claims)
+
+			untyped, err := WorkflowPrincipalFromIDToken(context.TODO(), token)
+			if err != nil {
+				if !test.WantErr {
+					t.Error(err)
+				}
+				return
+			}
+			if err == nil && test.WantErr {
+				t.Error("expected error")
+				return
+			}
+
+			gotName := untyped.Name(context.TODO())
```

Comment:
nit: Should this be moved to its own test, since it's testing a different function?